### PR TITLE
Add exception handlers to protect from bad Rx subscribers

### PR DESCRIPTION
**Pull request by tbak**

### CODE REVIEW

File: titus-common/src/main/java/com/netflix/titus/common/framework/reconciler/internal/FailedTransaction.java [hunk 1]
```diff
@@ -71,7 +76,11 @@ public boolean completeSubscribers() {
         if (transactionStep == TransactionStep.ErrorEventsEmitted) {
             this.transactionStep = TransactionStep.SubscribersCompleted;
 
-            changeActionHolder.getSubscriber().onError(error);
+            try {
+                changeActionHolder.getSubscriber().onError(error);
+            } catch (Exception e) {
+                logger.warn("Client subscriber onError handler thrown an exception: rootId={}, error={}", engine.getRunningView().getId(), e.getMessage());
```
Comment (corindwyer) [inline titus-common/src/main/java/com/netflix/titus/common/framework/reconciler/internal/FailedTransaction.java]: thrown -> threw

File: titus-common/src/main/java/com/netflix/titus/common/framework/reconciler/internal/SingleTransaction.java [hunk 2]
```diff
@@ -198,11 +198,19 @@ public void emitEvents() {
     @Override
     public boolean completeSubscribers() {
         if (transactionStep == TransactionStep.EventsEmitted) {
-            changeActionSubscriber.ifPresent(Observer::onCompleted);
+            try {
+                changeActionSubscriber.ifPresent(Observer::onCompleted);
+            } catch (Exception e) {
+                logger.warn("Client subscriber onCompleted handler thrown an exception: rootId={}, transactionId={}, error={}", engine.getRunningView().getId(), transactionId, e.getMessage());
```
Comment (corindwyer) [inline titus-common/src/main/java/com/netflix/titus/common/framework/reconciler/internal/SingleTransaction.java]: thrown -> threw

File: titus-common/src/main/java/com/netflix/titus/common/framework/reconciler/internal/SingleTransaction.java [hunk 3]
```diff
@@ -198,11 +198,19 @@ public void emitEvents() {
...
+            }
             this.transactionStep = TransactionStep.SubscribersCompleted;
             return true;
         } else if (transactionStep == TransactionStep.ErrorEventsEmitted) {
-            changeActionSubscriber.ifPresent(subscriber -> subscriber.onError(changeActionError));
+            try {
+                changeActionSubscriber.ifPresent(subscriber -> subscriber.onError(changeActionError));
+            } catch (Exception e) {
+                logger.warn("Client subscriber onError handler thrown an exception: rootId={}, transactionId={}, error={}", engine.getRunningView().getId(), transactionId, e.getMessage());
```
Comment (corindwyer) [inline titus-common/src/main/java/com/netflix/titus/common/framework/reconciler/internal/SingleTransaction.java]: thrown -> threw

File: titus-common/src/main/java/com/netflix/titus/common/framework/reconciler/internal/FailedTransaction.java (hunk 1 above)
Comment (tbak) [reply titus-common/src/main/java/com/netflix/titus/common/framework/reconciler/internal/FailedTransaction.java]: Fixed

File: titus-common/src/main/java/com/netflix/titus/common/framework/reconciler/internal/SingleTransaction.java (hunk 2 above)
Comment (tbak) [reply titus-common/src/main/java/com/netflix/titus/common/framework/reconciler/internal/SingleTransaction.java]: Fixed

File: titus-common/src/main/java/com/netflix/titus/common/framework/reconciler/internal/SingleTransaction.java (hunk 3 above)
Comment (tbak) [reply titus-common/src/main/java/com/netflix/titus/common/framework/reconciler/internal/SingleTransaction.java]: Fixed

**Pull request merged**